collector/lib/CollectionMethod.cpp

```diff
@@ -13,4 +15,10 @@ std::ostream& operator<<(std::ostream& os, CollectionMethod method) {
   }
 }
 
+std::string CollectionMethodName(CollectionMethod method) {
```

Comment:
I feel like we should have this the other way around, `CollectionMethodName` should check the `CollectionMethod` value and return a `const char*`, then the `operator<<` can just call into this function and any other code requiring a string can just build it, WDYT? It would save us some stream shenanigans.

For the unknown case, realistically, we should never get that, so just using `"unknown"` would be enough IMO (this wouldn't be a problem in Rust BTW).

Reply:
yeah seems very reasonable!